src/host.ps1

```diff
@@ -0,0 +1,162 @@
+        }
+        
+        return $resultObject
+    }
+}
+
+function Get-XoHost {
+    <#
+    .SYNOPSIS
+        Get physical hosts from Xen Orchestra.
+    .DESCRIPTION
+        Retrieves physical XCP-ng/XenServer hosts from Xen Orchestra. 
+        Can retrieve specific hosts by their UUID or filter hosts by various criteria.
+    .PARAMETER HostId
+        The UUID(s) of the host(s) to retrieve.
+    .PARAMETER Filter
+        Filter to apply to the host query.
+    .PARAMETER Limit
+        Maximum number of results to return.
+    .EXAMPLE
+        Get-XoHost
+        Returns all hosts.
+    .EXAMPLE
+        Get-XoHost -HostId "12345678-abcd-1234-abcd-1234567890ab"
+        Returns the host with the specified UUID.
+    .EXAMPLE
+        Get-XoHost -Filter "power_state:running"
+        Returns all running hosts.
+    #>
+    [CmdletBinding(DefaultParameterSetName = "All")]
+    param(
+        [Parameter(Mandatory, ValueFromPipelineByPropertyName, Position = 0, ParameterSetName = "HostId")]
+        [ValidatePattern("[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}")]
+        [Alias("HostUuid")]
+        [string[]]$HostId,
+        
+        [Parameter(ParameterSetName = "Filter")]
+        [string]$Filter,
+        
+        [Parameter(ParameterSetName = "Filter")]
+        [Parameter(ParameterSetName = "All")]
+        [int]$Limit
+    )
+
+    begin {
+        Write-Verbose "Getting XO hosts"
+        $params = @{}
+        if ($PSBoundParameters.ContainsKey('Filter')) {
+            $params['filter'] = $Filter
+        }
+        if ($PSBoundParameters.ContainsKey('Limit')) {
+            $params['limit'] = $Limit
+        }
+    }
+
+    process {
+        if ($PSCmdlet.ParameterSetName -eq "HostId") {
+            foreach ($id in $HostId) {
+                try {
+                    Write-Verbose "Getting host with ID $id"
+                    $hostData = Invoke-RestMethod -Uri "$script:XoHost/rest/v0/hosts/$id" @script:XoRestParameters
+                    if ($hostData) {
+                        ConvertTo-XoHostObject $hostData
+                    }
+                } catch {
+                    Write-Error "Failed to retrieve host with ID $id. Error: $_"
+                }
+            }
+        }
+    }
+    
+    end {
+        if ($PSCmdlet.ParameterSetName -eq "All" -or $PSCmdlet.ParameterSetName -eq "Filter") {
+            try {
+                Write-Verbose "Getting all hosts" 
+                $hostUrls = Invoke-RestMethod -Uri "$script:XoHost/rest/v0/hosts" @script:XoRestParameters
+                
+                if ($hostUrls -and $hostUrls.Count -gt 0) {
+                    Write-Verbose "Found $($hostUrls.Count) hosts"
+                    
+                    # Apply limit if specified
+                    $processLimit = if ($Limit -gt 0) { [Math]::Min($Limit, $hostUrls.Count) } else { $hostUrls.Count }
+                    $processUrls = $hostUrls | Select-Object -First $processLimit
```

Comment:
@iButcat should use the `limit` query parameter in XO; same for all other fetches